torch/csrc/jit/tensorexpr/cuda_codegen.cpp

```diff
@@ -855,15 +855,26 @@ void CudaCodeGen::call(const std::vector<CallArg>& args) {
 
   std::vector<int> gpu_block_extents_v(3, 1);
   std::vector<int> gpu_thread_extents_v(3, 1);
+
   // evaluate all the block/thread extents into values
   // TODO: eventually, codegen these calculations and make them part of the
   // module.
   for (size_t i = 0; i < gpu_block_extents.size(); i++) {
+    auto extent = dynamic_cast<const IntImm*>(gpu_block_extents[i]);
```

Comment:
It's almost certainly an int, but supporting more dtypes doesn't cost us much here:

```
if (gpu_block_extents[i]->isConstant()) {
  gpu_block_extents_v[i] = immeditateAs<int>(gpu_block_extents[i]);
}
```